lib/fencing/st_client.c

```diff
@@ -2619,6 +2619,40 @@ stonith__sort_history(stonith_history_t *history)
     return new;
 }
 
+stonith_history_t *
+stonith__first_matching_event(stonith_history_t *history,
+                              bool (*matching_fn)(stonith_history_t *, void *),
+                              void *user_data)
+{
+    for (stonith_history_t *hp = history; hp; hp = hp->next) {
+        if (matching_fn(hp, user_data)) {
+            return hp;
+        }
+    }
+
+    return NULL;
+}
+
+bool
+stonith__event_state_pending(stonith_history_t *history, void *user_data)
+{
+    return history->state != st_failed && history->state != st_done;
+}
+
+bool
+stonith__event_state_eq(stonith_history_t *history, void *user_data)
+{
+    int *state = (int *) user_data;
+    return history->state == *state;
```

Comment:
No need to change, but FYI you can also GPOINTER_TO_INT() and GINT_TO_POINTER() to pass the int directly rather than its pointer (the advantage being you could pass a constant as well as a variable, but OTOH it's probably a little uglier)

Reply:
Eh, I think that's a reasonable change to make.